front-end/src/components/newAccountForm.tsx

```diff
@@ -0,0 +1,188 @@
+import * as React from "react";
+import { ChangeEvent, useState } from "react";
+import {
+  Alert,
+  Link,
+  FormGroup,
+  Snackbar,
+  TextField,
+} from "@mui/material";
+import { Button } from "@mui/material";
+import history from "./history";
+
+export interface newAcccountFormProps {}
+
+export function NewAccountForm(props: newAcccountFormProps) {
+  const styles = {
+    signIn: {
+      display: "flex",
+      flexDirection: "column",
+      backgroundColor: "rgba(255,255,255, 0.9)",
+      borderRadius: "8px",
+      padding: "25px 75px",
+    } as const,
+    input: {
+      margin: "10px",
+    },
+    button: {
+      backgroundColor: "#D64309",
+    },
+  };
+
+  const [firstNameField, setFirstNameField] = useState("");
+  const [lastNameField, setLastNameField] = useState("");
+  const [emailField, setEmailField] = useState("");
+  const [passwordField, setPasswordField] = useState("");
+  const [passwordField2, setPasswordField2] = useState("");
+  const [alertReason, setAlertReason] = useState("");
+
+  const handleFirstNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setFirstNameField(e.target.value);
+  };
+
+  const handleLastNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setLastNameField(e.target.value);
+  };
+
+  const handleEmailChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setEmailField(e.target.value);
+  };
+
+  const handlePasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setPasswordField(e.target.value);
+  };
+
+  const handlePassword2Change = (e: ChangeEvent<HTMLInputElement>) => {
+    setPasswordField2(e.target.value);
+  };
+
+  //Alpha Characters only (case insensitive)
+  const checkSignage = /^[a-z]+$/i;
+
+  //Basic email signage.
+  const checkEmailSignage = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i
+
+  /* min eight characters,
+  max 20 characters, 
+  at least one letter, one number 
+  and one special character */
+  const checkPassSignage = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&^])[A-Za-z\d@$!%*#?&^]{8,20}$/;
+
+  const handleSubmit = () => {
+    console.log(`First Name: ${firstNameField}`);
+    console.log(`Last Name: ${lastNameField}`);
+    console.log(`Email: ${emailField}`);
+    console.log(`Password: ${passwordField}`);
+    console.log(`Check Password: ${passwordField2}`);
+
+    //All fields must be filled
+    if (firstNameField  === "" || lastNameField  === "" || emailField === "" || passwordField === "" || passwordField2 === "") {
+      setAlertReason("Enter your name, email address, and password.");
+      return;
+    }
+
+    //Check username/password requirements and security for text fields.
+    //Firstname and Lastname forms must use appropriate signage (alpha characters only)
+    if (!(firstNameField.match(checkSignage)) || !(lastNameField.match(checkSignage)) ) {
```

Comment:
Maybe modify this to allow hyphens, and maybe apostrophes? We want Joseph Gordon-Levitt  to be able to use our site. 😂

Reply:
Will do.

Reply:
Just made a commit for this.